### PR TITLE
feat: add collections to httpshopify

### PR DESCRIPTION
**Pull request by JamieMohc**
add collections to httpshopify

### CODE REVIEW

File: collections.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/MOHC-LTD/httpshopify/internal/http"
+	"github.com/MOHC-LTD/shopify"
+)
+
+type CollectionRepository struct {
```
Comment (harry-mohc) [inline collections.go:12]: All exported things need comments

File: collections.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+func newCollectionRepository(client http.Client, createURL func(endpoint string) string) CollectionRepository {
+	return CollectionRepository{
+		client,
+		createURL,
+	}
+}
+
+func (repository CollectionRepository) Get(ID int64) (shopify.Collection, error) {
+
```
Comment (harry-mohc) [inline collections.go:25]: no need for this whitespace

File: collections.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	url := repository.createURL(fmt.Sprintf("collections/%v.json%v", ID, parseCollectionQuery()))
```
Comment (harry-mohc) [inline collections.go:26]: this get doesnt take a query so you dont need to parse one here

File: collections.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	json.Unmarshal(body, &resultDTO)
+
+	return resultDTO.Collection.ToShopify(), nil
+}
+
+func (repository CollectionRepository) Products(ID int64) (shopify.Products, error) {
+
+	url := repository.createURL(fmt.Sprintf("collections/%v/products.json%v", ID, parseCollectionQuery()))
```
Comment (harry-mohc) [inline collections.go:44]: Looks like this endpoint is paginated so you'll need to handle that, also dont need the query

File: collections.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+		PublishedAt:    dto.PublishedAt,
+		PublishedScope: dto.PublishedScope,
+		Image:          dto.Image.ToShopify(),
+		SortOrder:      dto.SortOrder,
+		TemplateSuffix: dto.TemplateSuffix,
+		UpdatedAt:      dto.UpdatedAt,
+	}
+}
+
+func parseCollectionQuery() string {
```
Comment (harry-mohc) [inline collections.go:103]: this isnt needed

**Review by harry-mohc — CHANGES_REQUESTED** [review]

**Comment by harry-mohc** [comment]
looks like you've got some conflicts

File: collections.go [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+func newCollectionRepository(client http.Client, createURL func(endpoint string) string) collectionRepository {
+	return collectionRepository{
+		client,
+		createURL,
+	}
+}
+
+func (repository collectionRepository) Get(ID int64) (shopify.Collection, error) {
```
Comment (harry-mohc) [inline collections.go:24]: `ID` should be `id`

File: collections.go [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+	var resultDTO struct {
+		Collection CollectionDTO `json:"collection"`
+	}
+
+	json.Unmarshal(body, &resultDTO)
+
+	return resultDTO.Collection.ToShopify(), nil
+}
+
+func (repository collectionRepository) Products(ID int64) (shopify.Products, error) {
```
Comment (harry-mohc) [inline collections.go:41]: `ID` should be `id`

File: collections.go [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+	ID             int64     `json:"id"`
+	Title          string    `json:"title"`
+	BodyHTML       string    `json:"body_html"`
+	Handle         string    `json:"handle"`
+	PublishedAt    time.Time `json:"published_at"`
+	PublishedScope string    `json:"published_scope"`
+	SortOrder      string    `json:"sort_order"`
+	Image          ImageDTO  `json:"image"`
+	TemplateSuffix string    `json:"template_suffix"`
+	UpdatedAt      time.Time `json:"updated_at"`
```
Comment (harry-mohc) [inline collections.go:83]: please can you make sure these are in the same order as in the shopify documentation?

File: images.go [hunk 9]
```diff
@@ -0,0 +1,27 @@
+package httpshopify
+
+import (
+	"time"
+
+	"github.com/MOHC-LTD/shopify"
+)
+
+// ImageDTO represents a Shopify Image
```
Comment (harry-mohc) [inline images.go:9]:
```suggestion
// ImageDTO represents a Shopify Image in HTTP requests and responses
```

File: shop.go [hunk 10]
```diff
@@ -97,3 +99,8 @@ func (shop Shop) Products() shopify.ProductRepository {
 func (shop Shop) InventoryLevels() shopify.InventoryLevelRepository {
 	return shop.inventoryLevels
 }
+
+// Collections returns an http implementation of a shopify collection repository
```
Comment (harry-mohc) [inline shop.go:103]:
```suggestion
// Collections returns an HTTP implementation of a Shopify collection repository
```

**Review by harry-mohc — CHANGES_REQUESTED** [review]

File: shop.go (hunk 10 above)
Comment (JamieMohc) [reply shop.go:103]: this comment is following the rest of the shop methods shall I change those as well then?

Comment (harry-mohc) [reply shop.go:103]: yes please

**Review by harry-mohc — APPROVED** [review]

**Pull request merged**